### ISSUE TITLE
AWS::EC2::FlowLog incorect check with ResourceType TransitGateway -> TrafficType should be ignored

### ISSUE THREAD

**Issue by AndreasAugustin** [aws-cloudformation/cfn-lint#2485]
### CloudFormation Lint Version

cfn-lint 0.71.0

### What operating system are you using?

Mac

### Describe the bug

Hi dear AWS cfn-lint team,

thanks for the nice linter.

With the checks for **AWS::EC2::FlowLogs** it seems that there is an issue when the **ResourceType** is set to **TransitGateway**.

Related to docs it states

```yaml
TrafficType
The type of traffic to monitor (accepted traffic, rejected traffic, or all traffic). This parameter is not supported for transit gateway resource types. It is required for the other resource types.

Required: No

Type: String

Allowed values: ACCEPT | ALL | REJECT
```
[^flow-logs-traffic-type]

With that setting the **TrafficType** property must not be added to the template but is checked in the cfn-linter.

[^flow-logs-traffic-type]: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-flowlog.html#aws-resource-ec2-flowlog-properties

[ec2-flow-logs]: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-flowlog.html#cfn-ec2-flowlog-logformat

### Expected behavior

When the **ResourceType: TransitGateway** property is set, the expected behaviour is that you get an error if **TrafficType** is set and not like currently if it is not set.

### Reproduction template

```yaml
MyFlowLog:
  Type: AWS::EC2::FlowLog
  Properties:
    DeliverLogsPermissionArn: !GetAtt FlowLogRole.Arn
    LogGroupName: FlowLogsGroup
    ResourceId: !Ref MyTransitGateway
    ResourceType: TransitGateway
    TrafficType: ALL
```

does not throw but 

```yaml
MyFlowLog:
  Type: AWS::EC2::FlowLog
  Properties:
    DeliverLogsPermissionArn: !GetAtt FlowLogRole.Arn
    LogGroupName: FlowLogsGroup
    ResourceId: !Ref MyTransitGateway
    ResourceType: TransitGateway
```

throws. This behavior must be switched.

Update:
The same is true for ResourceType: TransitgatewayAttachement

**Comment by AndreasAugustin** [comment]
Remark:
The same is true for ResourceType: TransitgatewayAttachement

**Comment by PatMyron** [comment]
Unable to reproduce `cfn-lint` throwing for either `TransitGateway` or `TransitGatewayAttachment` with the current version with or without `TrafficType`
